neos/neos-development-collection
# BUGFIX: fixed black shapes in pdf thumbnails

**What I did**
Tested with the following imagick versions:

- V. 6.7.8
- V. 6.9
- V. 7.0

**How I did it**
See below

**How to verify it**
You can verifiy it by running this small code local:

```php
<?php
$filePath = 'pdf.pdf[0]';
$width = 500;
$height = 500;

$im = new \Imagick();
$im->setResolution(120, 120);
$im->readImage($filePath);
$im->setImageFormat('png');
$im->setImageBackgroundColor('white');
$im->setImageCompose(\Imagick::COMPOSITE_OVER);

if (method_exists($im, 'mergeImageLayers')) {
    // Replace flattenImages in imagick 3.3.0
    // @see https://pecl.php.net/package/imagick/3.3.0RC2
    $im = $im->mergeImageLayers(\Imagick::LAYERMETHOD_MERGE);
} else {
    $im->flattenImages();
}

if (defined('\Imagick::ALPHACHANNEL_OFF')) {
    // ImageMagick >= 7.0, Imagick >= 3.4.3RC1
    // @see https://pecl.php.net/package/imagick/3.4.3RC1
    $im->setImageAlphaChannel(\Imagick::ALPHACHANNEL_OFF);
} else {
    $im->setImageAlphaChannel(\Imagick::ALPHACHANNEL_RESET);
}

$im->thumbnailImage($width, $height, true);

$im->writeImage('thumb.png');
$im->destroy();
```
**Checklist**

- ?? The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
